internal/installation/move.go

```diff
@@ -152,10 +152,10 @@ func moveAllFiles(fromDir, toDir string, fos []index.FileOperation) error {
 }
 
 // moveToInstallDir moves plugins from srcDir to dstDir (created in this method) with given FileOperation.
-func moveToInstallDir(srcDir, installDir string, fos []index.FileOperation) error {
-	klog.V(4).Infof("Creating plugin installation directory %q", installDir)
-	if err := os.MkdirAll(installDir, 0755); err != nil {
-		return errors.Wrapf(err, "error creating installation directory at %q", installDir)
+func moveToInstallDir(srcDir, pluginDir, installDir string, fos []index.FileOperation) error {
+	klog.V(4).Infof("Creating plugin directory %q", pluginDir)
```

Comment:
I suspect this is not the right solution.
We don't really need the knowledge of PluginInstallPath in this method. We just need to know installDir.

Reply:
Can you try removing this os.MkdirAll here? I'm curious what'll happen. It might work.

Reply:
That's sort of how I felt as well, I was going back and forth between this and something like removing the version part from the `installDir` string that gets passed to mkdir.

Reply:
I did remove it, and posted the output in the description of the PR. It has this error:
>W1227 11:06:26.208294 6026 install.go:131] failed to install plugin "grep": install failed: failed while moving files to the installation directory: could not rename/copy directory "/tmp/krew-temp-move705873968" to "/home/kimc/playground/store/grep/v1.2.2": rename /tmp/krew-temp-move705873968 /home/kimc/playground/store/grep/v1.2.2: no such file or directory

Its trying to move a directory to `.../store/grep/v1.2.2` but `.../store/grep` doesn't exist without the call to mkdir.

Reply:
I think we actually need to do this:

    os.MkdirAll(filepath.Dir(installDir))

we are essentially trying to make sure installDir can be `mkdir`'ed further down in the call stack.

I think this would work, we don't need to pass `pluginDir` knowledge here (despite they're the same value, at least today).

Reply:
Ah that's a good idea, I didn't know about that method. I'll make the change.